### PR TITLE
New helpers command functionality

### PR DESCRIPTION
**Pull request by sbillinge**
@scopatz this is nearly ready for review but I want to do one more thing before I unWIP it.

It implements regolith "helpers" which are supposed to be lighter and easier to implement than full builders. They can be for things like making todo's inserting new people into the people collection and so on.  I have been putting these in my databases local directory, but it is time to start adding them to the project as a whole.  Also, they were just reading and editing yaml but I want them to make use of hte full regolith framework.

### CODE REVIEW

**Comment by sbillinge** [comment]
@scopatz this should be ready for review and merge now

**Comment by sbillinge** [comment]
@scopatz @CJ-Wright I would like to merge this as I have a bunch of activities that will use this.  Would it be possible for you to glance over it?  If you need more time to look it over please let me know.  Otherwise, if I don't hear I will go ahead and merge it tomorrow.  Once it is merged I will add more and more "helpers" which I can merge myself without bothering you, but since this does move regolith functionality in a new direction I would rather get your green light.

File: regolith/helper.py [hunk 1]
```diff
@@ -0,0 +1,14 @@
+"""Generic builder."""
+
+from regolith.helpers.hellohelper import HelloHelper, subparser
+from regolith.helpers import hellohelper as hello
+
+HELPERS = {
+    "hello": (hello.HelloHelper, hello.subparser)
+#    "helper": HelperBuilder,
```
Comment (CJ-Wright) [inline regolith/helper.py:8]:
```suggestion
```

File: regolith/helpers/basehelper.py [hunk 2]
```diff
@@ -0,0 +1,156 @@
...
+        super().__init__(rc)
+        self.cmds = ["sout"]
+
+class LatexHelperBase(HelperBase):
+    """Base class for Latex builders"""
+
+    def __init__(self, rc):
+        super().__init__(rc)
+        self.cmds = ["latex", "clean"]
+#        if HAVE_BIBTEX_PARSER:
```
Comment (CJ-Wright) [inline regolith/helpers/basehelper.py:108]: I'd remove the commented out code

File: regolith/helpers/hellohelper.py [hunk 3]
```diff
@@ -0,0 +1,71 @@
...
+    filter_grants,
+    fuzzy_retrieval,
+)
+
+def subparser(subpi):
+    subpi.add_argument("--person", help="pi first name space last name in quotes",
+                        default=None)
+    return subpi
+
+class HelloHelper(SoutHelperBase):
```
Comment (CJ-Wright) [inline regolith/helpers/hellohelper.py:21]: Is this a descriptive name for what this is doing?

File: regolith/helpers/hellohelper.py [hunk 4]
```diff
@@ -0,0 +1,71 @@
...
+    """Build a helper"""
```
Comment (CJ-Wright) [inline regolith/helpers/hellohelper.py:22]: This isn't descriptive

File: regolith/helpers/testhelper.py [hunk 5]
```diff
@@ -0,0 +1,65 @@
...
+from regolith.dates import month_to_int
+from regolith.fsclient import _id_key
+from regolith.sorters import position_key
+from regolith.tools import (
+    all_docs_from_collection,
+    filter_grants,
+    fuzzy_retrieval,
+)
+
+class HelloHelper(SoutHelperBase):
```
Comment (CJ-Wright) [inline regolith/helpers/testhelper.py:16]: Is this duplicate code?

File: tests/test_helpers.py [hunk 6]
```diff
@@ -0,0 +1,56 @@
...
+import pytest
+
+from regolith.broker import load_db
+from regolith.main import main
+
+import openpyxl
+
+helper_map = [
+    (["helper", "hello", "--person", "Simon"],"hello Simon\n")
+        #["a_proprev",
```
Comment (CJ-Wright) [inline tests/test_helpers.py:15]: do you still want these tests?

File: regolith/helpers/hellohelper.py (hunk 3 above)
Comment (CJ-Wright) [inline regolith/helpers/hellohelper.py:21]: Why is this a helper and not a builder?

**Comment by sbillinge** [comment]
> 
> 
Thanks for the comments CJ

> I'm not certain I understand the difference between builders and helpers based on the examples given. Also there seems to be some code duplication.

There is some overlap, but helpers will be able to update the databases and do lightweight "builder-like" tasks such as simply printing reports to stdout. 

From my perspective the most useful thing is that I can fill them with bespoke command line args to  fine-tune them very much without having to update main.py and making the subparser list there 5000 lines long.

File: tests/test_helpers.py (hunk 6 above)
Comment (sbillinge) [reply tests/test_helpers.py:15]: That line will delete when the proprev adder is merged

File: regolith/helpers/testhelper.py (hunk 5 above)
Comment (sbillinge) [reply regolith/helpers/testhelper.py:16]: ah, interesting.  Let me look at that...

File: regolith/helpers/basehelper.py (hunk 2 above)
Comment (sbillinge) [reply regolith/helpers/basehelper.py:108]: will do

**Comment by sbillinge** [comment]
@CJ-Wright I hopefully addressed all your comments and fixed all the things.

**Comment by sbillinge** [comment]
I have all these commits now so I am closing.

**Pull request closed without merging**